src/main/java/software/amazon/cloudwatchlogs/emf/model/MetricDefinitionBuilder.java

```diff
@@ -0,0 +1,123 @@
+/*
+ *   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package software.amazon.cloudwatchlogs.emf.model;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import lombok.Getter;
+import lombok.NonNull;
+
+/** Represents the MetricDefinition of the EMF schema. */
+class MetricDefinition extends Metric {
+    @JsonIgnore @NonNull @Getter protected List<Double> values;
+
+    MetricDefinition(Unit unit, StorageResolution storageResolution, List<Double> values) {
+        this.unit = unit;
+        this.storageResolution = storageResolution;
+        this.values = values;
+    }
+
+    MetricDefinition(
+            String name, Unit unit, StorageResolution storageResolution, List<Double> values) {
+        this.unit = unit;
+        this.storageResolution = storageResolution;
+        this.values = values;
+        this.name = name;
+    }
+
+    MetricDefinition(String name, Unit unit, StorageResolution storageResolution, Double value) {
+        this.unit = unit;
+        this.storageResolution = storageResolution;
+        this.values = Arrays.asList(value);
+        this.name = name;
+    }
+
+    @Override
+    protected Metric getMetricValuesUnderSize(int size) {
+        List<Double> subList = values.subList(0, size > values.size() ? values.size() : size);
+        return new MetricDefinition(name, unit, storageResolution, subList);
+    }
+
+    @Override
+    protected Metric getMetricValuesOverSize(int size) {
+        if (size > values.size()) {
+            return null;
+        }
+        List<Double> subList = values.subList(size, values.size());
+        return new MetricDefinition(name, unit, storageResolution, subList);
+    }
+
+    /**
+     * @return the values of this metric, simplified to a double instead of a list if there is only
+     *     one value
+     */
+    @Override
+    protected Object getFormattedValues() {
+        return values.size() == 1 ? values.get(0) : values;
+    }
+}
+
+/** Builds MetricDefinition */
+public class MetricDefinitionBuilder extends MetricDefinition implements MetricBuilder {
```

Comment:
What about using lombok's `@Builder` instead?

Reply:
@Builder does not work well with the inheritance being used here and complicates the code since each builder needs to extend the class it is building